attributejson.go

```diff
@@ -0,0 +1,56 @@
+package yoti
+
+import (
+	"log"
+)
+
+//AttributeJSON is a Yoti attribute which returns a string as its value
+type AttributeJSON struct {
+	attribute
+}
+
+func newAttributeJSON(byteValue []byte, anchors []Anchor, name string, attrType AttrType) (result AttributeJSON) {
+	if attrType != AttrTypeJSON {
+		log.Printf("Cannot create JSON attribute with non-JSON type: %q", attrType.String())
+		return
```

Comment:
Again not really sure this in inline with go "best practices", usually if a function has the possibility of not doing what it's supposed to (due to bad input or transient error), it should "signal" that through it's interface by returning `err`. One of the "things" in go is to make error handling mandatory (by returning `err` in function where stuff can go wrong).

Reply:
In the SDKs we've taken the stance to not fail the attribute parsing process if an error occurs, so that the other attributes can still be processed, instead we are just logging the error. Given this, would returning an err at this stage provide us with any other benefits?

Reply:
The question is who/what decides to continue on, presumably there is a "main" method that is parsing all the attributes. The further down the stack, the less opinionated the code should be in regards to "business" logic and error handling.

What I mean is the lower layers should return errors and then at some point  there will be a "layer" that is controlling the main flow of a certain action where all "business" rules will be will be enforced and will decide whether to abort or to continue on in case of error (or in case of a particular error). Even the logging decision, usually you either log or you propagate an error, but not both.

I'm not too into design decisions here so consider this comment only food for though :wink: but from a design perspective it makes for cleaner code if there is a single error handling approach: always propagate err, until you can't or until you reach code that "owns" the business behavior (and enforces it).